widgetssdk/src/test/java/com/glia/widgets/callvisualizer/ActivityWatcherControllerTest.kt

```diff
@@ -0,0 +1,256 @@
+    private val screenSharingController = mock(ScreenSharingController::class.java)
+    private val isCallOrChatActiveUseCase = mock(IsCallOrChatScreenActiveUseCase::class.java)
+    private val watcher = mock(ActivityWatcherContract.Watcher::class.java)
+    private val controller = ActivityWatcherController(callVisualizerController, screenSharingController, serviceChatHeadController)
+    private val activity = mock(Activity::class.java)
+    private val activityReference: WeakReference<Activity?> = WeakReference(activity)
+    private val view = mock(View::class.java)
+
+    @Before
+    fun setup() {
+        val window = mock(Window::class.java)
+        val view = mock(View::class.java)
+        whenever(activity.window).thenReturn(window)
+        whenever(window.decorView).thenReturn(view)
+        controller.setWatcher(watcher)
+        cleanup()
+        resetMocks()
+    }
+
+    @Test
+    fun `onActivityResumed screen sharing skipped when call or chat is active`() {
+        whenever(callVisualizerController.isCallOrChatScreenActiveUseCase).thenReturn(isCallOrChatActiveUseCase)
+        whenever(isCallOrChatActiveUseCase(activity)).thenReturn(true)
+        whenever(watcher.fetchGliaOrRootView()).thenReturn(view)
+        controller.onActivityResumed(activityReference)
+        verify(watcher).fetchGliaOrRootView()
+        verify(callVisualizerController, times(2)).isCallOrChatScreenActiveUseCase
+        verify(serviceChatHeadController).onResume(view)
+    }
+
+    @Test
+    fun `onActivityResumed bubble is resumed when onScreenSharingStarted`() {
+        `onActivityResumed callbacks are set when call or chat are not active`()
+        whenever(watcher.fetchGliaOrRootView()).thenReturn(view)
+        controller.screenSharingViewCallback?.onScreenSharingStarted()
+        verify(watcher).fetchGliaOrRootView()
+        verify(serviceChatHeadController).onResume(view)
+    }
+
+    @Test
+    fun `onActivityResumed error is shown when onScreenSharingError`() {
+        `onActivityResumed callbacks are set when call or chat are not active`()
+        controller.screenSharingViewCallback?.onScreenSharingRequestError(GliaException("message", GliaException.Cause.INTERNAL_ERROR))
+        verify(watcher).showToast("message")
+    }
+
+    @Test
+    fun `onActivityPaused cleanup of callbacks`() {
+        whenever(watcher.fetchGliaOrRootView()).thenReturn(view)
+        controller.onActivityPaused()
+        verify(watcher).dismissAlertDialog(false)
+        verify(watcher).removeDialogCallback()
+        verify(watcher).fetchGliaOrRootView()
+        verify(screenSharingController).removeViewCallback(anyOrNull())
+        verify(serviceChatHeadController).onPause(view)
+    }
+
+    @Test
+    fun `onActivityDestroyed bubble is destroyed`() {
+        controller.onActivityDestroyed()
+        verify(serviceChatHeadController).onDestroy()
+    }
+
+    @Test
+    fun isCallOrChatActive() {
```

Comment:
I think the name of this test doesn't follow the naming pattern.

Reply:
fixed